app/src/main/java/com/padc/interactive_training/views/holders/ChapterViewHolder.java

```diff
@@ -0,0 +1,136 @@
+
+    @BindView(R.id.iv_lock)
+    ImageView ivLock;
+
+    @BindView(R.id.tv_lock_label)
+    TextView tvLock;
+
+    @BindView(R.id.tv_duration)
+    TextView tvDuration;
+
+    @BindView(R.id.tv_card_count)
+    TextView tvCardCount;
+
+    @BindView(R.id.tv_finished_percentage)
+    TextView tvFinishedPercentage;
+
+    @BindView(R.id.view_finished_percentage)
+    View vFinished;
+
+    @BindView(R.id.view_remaining_percentage)
+    View vRemaining;
+
+    private Context mContext;
+
+    private ChapterVO mChapterVO;
+    private ControllerChapterItem mController;
+    private View mSelfView;
+
+    public ChapterViewHolder(View itemView, ControllerChapterItem controller) {
+        super(itemView);
+        ButterKnife.bind(this, itemView);
+
+        this.mContext = InteractiveTrainingApp.getContext();
+        this.mController = controller;
+        this.mSelfView = itemView;
+    }
+
+    public void bindData(ChapterVO chapter) {
+        this.mChapterVO = chapter;
+
+        tvChapterNumber.setText("Chapter-" + String.valueOf(chapter.getChapterNumber()));
+        tvChapterTitle.setText(chapter.getTitle());
+        tvChapterBrief.setText(chapter.getChapterBrief());
+
+        String durtionInMinute = tvDuration.getResources().getQuantityString(
+                R.plurals.minutes_count, chapter.getDurationInMins(), chapter.getDurationInMins()
+        );
+        tvDuration.setText(durtionInMinute);
+
+        String cardsCount = tvDuration.getResources().getQuantityString(
+                R.plurals.cards_count, chapter.getLessonCount(), chapter.getLessonCount()
+        );
+        tvCardCount.setText(cardsCount);
+
+        if (chapter.isLocked()) {
+            // Define Colors for Disabled State
+            tvChapterNumber.setTextColor(mContext.getResources().getColor(R.color.chapter_disabled_text_color));
+            tvChapterTitle.setTextColor(mContext.getResources().getColor(R.color.chapter_disabled_text_color));
+            tvChapterBrief.setTextColor(mContext.getResources().getColor(R.color.chapter_disabled_text_color));
+
+            // Hide Duration and Cards Count
+            tvDuration.setVisibility(View.INVISIBLE);
+            tvCardCount.setVisibility(View.INVISIBLE);
+            tvFinishedPercentage.setVisibility(View.INVISIBLE);
+
+            // Define Lock Disabled Color
+            vFinished.setBackgroundColor(mContext.getResources().getColor(R.color.chapter_disabled_background));
+            vFinished.setLayoutParams(new LinearLayout.LayoutParams(0,
+                    ViewGroup.LayoutParams.MATCH_PARENT, 100));
+
+            vRemaining.setLayoutParams(new LinearLayout.LayoutParams(0,
+                    ViewGroup.LayoutParams.MATCH_PARENT, 0));
+
+        } else {
+            ivLock.setVisibility(View.GONE);
+            tvLock.setVisibility(View.GONE);
+
+            // Define Finished Percentage if it's not Lock
+            vFinished.setLayoutParams(new LinearLayout.LayoutParams(0,
```

Comment:
I'm not sure but are you trying to replicate the effect of ProgressBar? If yes, there's library like [this](https://github.com/akexorcist/Android-RoundCornerProgressBar) to help you